brokerapi/brokers/pubsub/broker.go

```diff
@@ -47,79 +46,72 @@ type InstanceInformation struct {
+	}
+
+	subscriptionConfig := googlepubsub.SubscriptionConfig{
+		PushConfig: googlepubsub.PushConfig{
+			Endpoint: endpoint,
+		},
+		AckDeadline: time.Duration(variableContext.GetInt("ack_deadline")) * time.Second,
+		Labels:      defaultLabels,
 	}
 
-	// Ensure there is a name for this topic
-	if _, ok := params["topic_name"]; !ok {
-		params["topic_name"] = name_generator.Basic.InstanceName()
+	if err := variableContext.Error(); err != nil {
+		return models.ServiceInstanceDetails{}, err
 	}
 
+	// Create
 	pubsubClient, err := b.createClient(ctx)
 	if err != nil {
 		return models.ServiceInstanceDetails{}, err
 	}
 
-	t, err := pubsubClient.CreateTopic(ctx, params["topic_name"])
+	topic, err := pubsubClient.CreateTopic(ctx, topicName)
 	if err != nil {
-		return models.ServiceInstanceDetails{}, fmt.Errorf("Error creating new pubsub topic: %s", err)
+		return models.ServiceInstanceDetails{}, fmt.Errorf("Error creating new Pub/Sub topic: %s", err)
 	}
 
-	i := models.ServiceInstanceDetails{
-		Name:         params["topic_name"],
-		Url:          "",
-		Location:     "",
-		OtherDetails: "{}",
-	}
-	ii := InstanceInformation{
-		TopicName: params["topic_name"],
+	// This service needs labels to be set after creation
+	if _, err := topic.Update(ctx, googlepubsub.TopicConfigToUpdate{Labels: defaultLabels}); err != nil {
+		return models.ServiceInstanceDetails{}, fmt.Errorf("Error setting labels on new Pub/Sub topic: %s", err)
 	}
 
-	if subscriptionName, ok := params["subscription_name"]; ok {
-		var pushConfig googlepubsub.PushConfig
-		var ackDeadline = 10
+	if subscriptionName != "" {
+		subscriptionConfig.Topic = topic
 
-		if ackd, ok := params["ack_deadline"]; ok {
-			ackDeadline, err = strconv.Atoi(ackd)
-			if err != nil {
-				return models.ServiceInstanceDetails{}, fmt.Errorf("Error converting ack deadline to int: %s", err)
-			}
-		}
-
-		if isPush, ok := params["is_push"]; ok {
-			if isPush == "true" {
-				pushConfig = googlepubsub.PushConfig{
-					Endpoint: params["endpoint"],
-				}
-			}
-		}
-
-		subsConfig := googlepubsub.SubscriptionConfig{
-			PushConfig:  pushConfig,
-			Topic:       t,
-			AckDeadline: time.Duration(ackDeadline) * time.Second,
-		}
-
-		_, err = pubsubClient.CreateSubscription(ctx, subscriptionName, subsConfig)
-		if err != nil {
+		if _, err := pubsubClient.CreateSubscription(ctx, subscriptionName, subscriptionConfig); err != nil {
 			return models.ServiceInstanceDetails{}, fmt.Errorf("Error creating subscription: %s", err)
 		}
+	}
 
-		ii.SubscriptionName = params["subscription_name"]
+	ii := InstanceInformation{
+		TopicName:        topicName,
+		SubscriptionName: subscriptionName,
 	}
 
 	otherDetails, err := json.Marshal(ii)
 	if err != nil {
 		return models.ServiceInstanceDetails{}, fmt.Errorf("Error marshalling json: %s", err)
 	}
-	i.OtherDetails = string(otherDetails)
 
-	return i, nil
+	return models.ServiceInstanceDetails{
+		Name:         topicName,
```

Comment:
It appears topicName is also not checked for an empty string before being used.

Reply:
This is semi-intetional. @mattysweeps is going to implement JSON Schema validation of user arguments some time this week, but I'll add the validation back in here so master is valid until then.

Reply:
Sounds good. Thanks.